bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\Date\Value;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\HandlerType;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+
+use function in_array;
+
+final class ScheduledVisibilityService
+{
+    public function __construct(
+        private readonly iterable $handlers,
+        private readonly string $type,
+        private readonly bool $enabled,
+        private readonly bool $allContentTypes,
+        private readonly array $allowedContentTypes,
+    ) {}
+
+    public function toggleVisibility(Content $content): void
+    {
+        $handlerType = HandlerType::from($this->type);
+
+        /** @var ScheduledVisibilityInterface $handler */
+        foreach ($this->handlers as $handler) {
+            if ($handler->getType() === $handlerType) {
+                if ($this->shouldHide($content)) {
+                    $handler->hide($content);
+
+                    return;
+                }
+                if ($this->shouldReveal($content)) {
+                    $handler->reveal($content);
+                }
+
+                return;
```

Comment:
There are 3 possible results here:

1. item was hidden
2. item was revealed
3. there was no change

Please add an Enum with these 3 options as the return type here.

Reply:
fixed in d4bc0d868fb0ee8eee7e9cd95c62fd231c9bf5e5